tests/helm/common_test.go

```diff
@@ -369,55 +371,65 @@ func isSubchartObject(object metav1.Object) bool {
 	return false
 }
 
-// Get a PodSpec from the unstructured object, if possible
-// This only works on Deployments, StatefulSets and DaemonSets
-func GetPodSpec(object unstructured.Unstructured) (*corev1.PodSpec, error) {
+func GetPodSpec(podTemplateSpec *corev1.PodTemplateSpec) *corev1.PodSpec {
+	if podTemplateSpec == nil {
+		return nil
+	}
+	return &podTemplateSpec.Spec
+}
+
+func GetPodTemplateSpec(object unstructured.Unstructured) (*corev1.PodTemplateSpec, error) {
```

Comment:
I would leave previous siganture
```suggestion
func GetPodTemplateSpec(object unstructured.Unstructured) (*corev1.PodTemplateSpec, error) {
	podTemplateSpec, err := GetPodTemplateSpec(object)
	if err != nil {
		return nil, err
	}
	return &podTemplateSpec.Spec, nil
}

func GetPodTemplateSpec(object unstructured.Unstructured) (*corev1.PodTemplateSpec, error) {
```

Reply:
It require less refactor, and we also keep checking for the `podTemplateSpec` inside the function